From c6fd85c58db4ae841dae0f7a8ba5602f29bc8d36 Mon Sep 17 00:00:00 2001
From: Klaus Zimmermann <klaus.zimmermann@quansight.com>
Date: Fri, 28 Feb 2025 13:25:58 +0100
Subject: [PATCH 1/2] Make cmake rocm detection required

---
 amd/hipcc/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/amd/hipcc/CMakeLists.txt b/amd/hipcc/CMakeLists.txt
index c7e08f685f13..5b74f678cf87 100755
--- a/amd/hipcc/CMakeLists.txt
+++ b/amd/hipcc/CMakeLists.txt
@@ -5,7 +5,7 @@ project(hipcc VERSION "1.1.1" LANGUAGES C CXX)
 include(CMakePackageConfigHelpers)
 include(GNUInstallDirs)
 
-find_package(ROCM QUIET)
+find_package(ROCM REQUIRED)
 if(ROCM_FOUND)
   include(ROCMSetupVersion)
   rocm_setup_version(VERSION "${hipcc_VERSION}")
-- 
2.43.0

