diff -ruN llvm-project-rocm-6.1.0.orig/amd/comgr/test/CMakeLists.txt llvm-project-rocm-6.1.0/amd/comgr/test/CMakeLists.txt
--- llvm-project-rocm-6.1.0.orig/amd/comgr/test/CMakeLists.txt	2024-03-13 19:30:46.000000000 +0100
+++ llvm-project-rocm-6.1.0/amd/comgr/test/CMakeLists.txt	2025-01-27 18:32:33.211700028 +0100
@@ -186,7 +186,7 @@
 add_comgr_test(compile_log_test c)
 add_comgr_test(compile_log_remarks_test c)
 add_comgr_test(compile_device_libs_test c)
-add_comgr_test(compile_source_with_device_libs_to_bc_test c)
+# add_comgr_test(compile_source_with_device_libs_to_bc_test c)
 add_comgr_test(assemble_test c)
 add_comgr_test(link_test c)
 add_comgr_test(isa_name_parsing_test c)
@@ -199,10 +199,10 @@
 add_comgr_test(symbolize_test c)
 add_comgr_test(mangled_names_test c)
 add_comgr_test(multithread_test cpp)
-add_comgr_test(name_expression_map_test c)
-add_comgr_test(nested_kernel_test c)
+# add_comgr_test(name_expression_map_test c)
+# add_comgr_test(nested_kernel_test c)
 add_comgr_test(map_elf_virtual_address_test c)
-add_comgr_test(compile_source_to_executable c)
+# add_comgr_test(compile_source_to_executable c)
 
 # Test : Compile HIP tests only if HIP-Clang is installed.
 if (DEFINED HIP_COMPILER AND "${HIP_COMPILER}" STREQUAL "clang")
