diff --git a/source/lib/output/generateJSON.hpp b/source/lib/output/generateJSON.hpp
index 4ffc78bd5..ed1d43e96 100644
--- a/source/lib/output/generateJSON.hpp
+++ b/source/lib/output/generateJSON.hpp
@@ -36,7 +36,7 @@ namespace rocprofiler
 {
 namespace tool
 {
-using JSONOutputArchive = ::cereal::MinimalJSONOutputArchive;
+using JSONOutputArchive = ::cereal::JSONOutputArchive;

 struct json_output
 {
diff --git a/source/lib/rocprofiler-sdk-tool/tool.cpp b/source/lib/rocprofiler-sdk-tool/tool.cpp
index 7afcbd9f7..ee47ee90b 100644
--- a/source/lib/rocprofiler-sdk-tool/tool.cpp
+++ b/source/lib/rocprofiler-sdk-tool/tool.cpp
@@ -2206,7 +2208,7 @@ struct output_data
 void
 generate_config_output(const tool::config& cfg, const tool::metadata& tool_metadata_v)
 {
-    using JSONOutputArchive = ::cereal::PrettyJSONOutputArchive;
+    using JSONOutputArchive = ::cereal::JSONOutputArchive;

     constexpr auto json_prec   = 16;
     constexpr auto json_indent = JSONOutputArchive::Options::IndentChar::space;